### PR TITLE
Integrated GitHub Authentication

### PR DESCRIPTION
**Pull request by lizardkingLK**
This pull request contains 

- GitHub Authentication
- Minor UI Tweaks
- VSCode launch.json config

### CODE REVIEW

**Comment by mfts** [comment]
Thabks for the submission. I'll have a look today :)

**Comment by lizardkingLK** [comment]
> Thanks for the submission. I'll have a look today :)

Thanks

**Comment by lizardkingLK** [comment]
@mfts hey is there a way this can get accepted?

File: components/ui/loader.tsx
Comment (mfts) [inline components/ui/loader.tsx:1]: This component is no longer required. we have `@/components/ui/loading-spinner`

File: pages/login.tsx [hunk 1]
```diff
@@ -52,6 +52,26 @@ export default function Login() {
...
+            className="flex justify-center items-center"
+          >
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              viewBox="0 0 16 16"
+              className="bi bi-github h-4 w-4 - mr-2"
+              fill="currentColor"
+            >
+              <path d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z" />
+            </svg>
```
Comment (mfts) [inline pages/login.tsx]: Please use the existing GitHub icon component: `@/components/shared/icons/github.tsx`

File: .vscode/launch.json
Comment (mfts) [inline .vscode/launch.json:1]: please exclude this file from the PR. it's your local preference but some other users may use a different editor or setup. :)

File: pages/documents/[id].tsx
Comment (mfts) [inline pages/documents/[id].tsx:1]: something with the formatting went wrong here. please revert changes on this file. it's not required to be edited for this PR. the exact error is that one of the div closes too early and there's an unclosed fragment. by reverting your changes it should fix this issue on this page

**Review by mfts — CHANGES_REQUESTED** [review]
Beside the inline comments, there are two other things that have come to my attentions. 

1. GitHub stores an additional field in the database [`refresh_token_expires_in`, which needs to be added to `Account` model in schema.prisma](https://next-auth.js.org/providers/github#:~:text=GitHub%20returns%20a%20field%20on%20Account%20called%20refresh_token_expires_in%20which%20is%20a%20number.%20See%20their%20docs.%20Remember%20to%20add%20this%20field%20to%20your%20database%20schema%2C%20in%20case%20if%20you%20are%20using%20an%20Adapter.)

2. secondly if the user has previously signup with another account with the same email, then it throws an **OAuthAccountNotLinked** error, which should be handled with a message saying something like "account with this email already exists, please use a different login method"

**Comment by lizardkingLK** [comment]
Thanks a lot. I will check.

**Pull request closed without merging**